### PR TITLE
Drop support for EOL rubies and rails

### PR DESCRIPTION
**Pull request by gmcgibbon**
Drops EOL support for dead rubies and rails versions. Also tests on latest minor versions.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,22 +2,20 @@ language: ruby
...
-  - 2.2.5
-  - 2.3.1
+  - 2.4.9
+  - 2.5.7
+  - 2.6.5
 
 gemfile:
-  - Gemfile.rails42
   - Gemfile.rails50
   - Gemfile.rails51
```
Comment (dylanahsmith) [inline .travis.yml]: Looks like these aren't supported either.  We could test 6.0 and rails master instead.

File: .travis.yml [hunk 2]
```diff
@@ -2,22 +2,20 @@ language: ruby
 sudo: false
 
 rvm:
-  - 2.2.5
-  - 2.3.1
+  - 2.4.9
+  - 2.5.7
+  - 2.6.5
```
Comment (dylanahsmith) [inline .travis.yml]: There is no ruby specific code, so it should be good enough to just test the minimum supported and latest release.  That is generally enough to make sure the library uses features that are available for all the supported versions.

The oldest ruby version isn't going to be compatible with the newest rails release, so we might just want to use a few of [named jobs](https://docs.travis-ci.com/user/customizing-the-build#naming-jobs-within-matrices) to cover minimum supported ruby & rails versions, latest released ruby and rails version and maybe a final one to cover rails master.

File: .travis.yml (hunk 1 above)
Comment (gmcgibbon) [reply .travis.yml]: Right, I think 5.2, 6.0, and head is good enough. 5.2 is still partially supported, see https://guides.rubyonrails.org/maintenance_policy.html.

Comment (dylanahsmith) [reply .travis.yml]: Yeah, I didn't mean to imply 5.2 wasn't supported.  This comment was only on the Gemfile.rails50 and Gemfile.rails51 lines

File: .travis.yml [hunk 3]
```diff
@@ -1,23 +1,13 @@
...
-  - Gemfile.rails51
-  - Gemfile.rails52
-
-matrix:
+jobs:
   include:
-    - rvm: 2.1.9
-      gemfile: Gemfile.rails42
-    - rvm: 2.5.0
+    - rvm: 2.4.9
```
Comment (dylanahsmith) [inline .travis.yml]: We should also be able to omit the bugfix release and get the latest bugfix release.  E.g.

```suggestion
    - rvm: 2.4
```

Comment (gmcgibbon) [reply .travis.yml]: Ah, right. Done!

Comment (gmcgibbon) [reply .travis.yml]: Actually, it seems to default to `2.4.6`. 🤔

Comment (dylanahsmith) [reply .travis.yml]: Oh well.  I don't think it matters enough to workaround unless it actually breaks the tests.

**Pull request merged**